### ISSUE TITLE
Option to use stored tokens

### ISSUE THREAD

**Issue by x1ddos** [x1ddos/simpleauth#15]
Reported by [muijsenbergq](https://code.google.com/u/112367194840494255839/)

---

Hi,

I'm using your library and i need to be able to use stored tokens and try to reuse them. As a separate path from the OAuth flow.

At the moment a _hacked_ away a little bit to achieve this functionality but maybe it's an idea to implement this (the right way)?

This i my method to use existing tokens and tap into your library.
I made "get_consumer_info_for()" a class method,made it always return a 3-tuple and for the rest using the implementation from your demo.

``` python
def try_oauth(tokens):
    provider = tokens.provider
    client_id, client_secret, scope = AuthHandler.get_consumer_info_for(provider)
    fetcher = getattr(AuthHandler(), '_get_%s_user_info' % provider)
    auth_info = {'oauth_token': tokens.oauth_token, 'oauth_token_secret': tokens.oauth_secret}
    user_data = fetcher(auth_info, key=client_id, secret=client_secret)
    if user_data:
        auth_id = get_auth_id(user_data, provider)
        ....
```

btw, using this class to store tokens in db

``` python
class UserTokens(db.Model):
    created = db.IntegerProperty(required=True)
    provider = db.StringProperty(indexed=False, required=True)
    cookie = db.StringProperty()
    oauth_token = db.StringProperty(indexed=False, required=True)
    oauth_secret = db.StringProperty(indexed=False, required=True)
    user = ....
```
## Thanks!

Moved here from https://code.google.com/p/gae-simpleauth/issues/detail?id=6

**Comment by chafreaky** [comment]
I'd actually be very interested in this enhancement. I am trying to pass a token from an iOS app to GAE using simpleauth (Facebook auth in the app, Open Graph call in App Engine) and I believe this would be a good step towards achieving what I am looking for.

**Comment by crhym3** [comment]
I'm not sure there's a real benefit in adding it as a feature to the library as OAuth 2.0 access tokens are normally short lived, unless you're using offline mode.

Do you have a demo or maybe a PR?

**Comment by chafreaky** [comment]
Good point, I think it might be better for me to just create a normal session on GAE and store the FB token as a session variable instead of going through the trouble of implementing FB Auth on the backend when the front-end already takes care of it.
